pkg/controller/operators/olm/operator_test.go

```diff
@@ -3706,35 +3757,40 @@ func TestSyncOperatorGroups(t *testing.T) {
 			assert.Equal(t, tt.expectedStatus, operatorGroup.Status)
 
 			for namespace, objects := range tt.final.objects {
-				for _, object := range objects {
-					var err error
-					var fetched runtime.Object
-					switch o := object.(type) {
-					case *appsv1.Deployment:
-						fetched, err = op.OpClient.GetDeployment(namespace, o.GetName())
-					case *rbacv1.ClusterRole:
-						fetched, err = op.OpClient.GetClusterRole(o.GetName())
-					case *rbacv1.Role:
-						fetched, err = op.OpClient.GetRole(namespace, o.GetName())
-					case *rbacv1.ClusterRoleBinding:
-						fetched, err = op.OpClient.GetClusterRoleBinding(o.GetName())
-					case *rbacv1.RoleBinding:
-						fetched, err = op.OpClient.GetRoleBinding(namespace, o.GetName())
-					case *v1alpha1.ClusterServiceVersion:
-						fetched, err = op.client.OperatorsV1alpha1().ClusterServiceVersions(namespace).Get(o.GetName(), metav1.GetOptions{})
-					case *v1.OperatorGroup:
-						fetched, err = op.client.OperatorsV1().OperatorGroups(namespace).Get(o.GetName(), metav1.GetOptions{})
-					default:
-						require.Failf(t, "couldn't find expected object", "%#v", object)
-					}
-					require.NoError(t, err, "couldn't fetch %s %v", namespace, object)
-					require.Equal(t, object, fetched, "%s in %s not equal", object.GetObjectKind().GroupVersionKind().String(), namespace)
-				}
+				RequireObjectsInNamespace(t, op.OpClient, op.client, namespace, objects)
 			}
 		})
 	}
 }
 
+func RequireObjectsInNamespace(t *testing.T, opClient operatorclient.ClientInterface, client versioned.Interface, namespace string, objects []runtime.Object) {
+	for _, object := range objects {
+		var err error
+		var fetched runtime.Object
+		switch o := object.(type) {
+		case *appsv1.Deployment:
+			fetched, err = opClient.GetDeployment(namespace, o.GetName())
+		case *rbacv1.ClusterRole:
+			fetched, err = opClient.GetClusterRole(o.GetName())
+		case *rbacv1.Role:
+			fetched, err = opClient.GetRole(namespace, o.GetName())
+		case *rbacv1.ClusterRoleBinding:
+			fetched, err = opClient.GetClusterRoleBinding(o.GetName())
+		case *rbacv1.RoleBinding:
+			fetched, err = opClient.GetRoleBinding(namespace, o.GetName())
+		case *v1alpha1.ClusterServiceVersion:
+			fetched, err = client.OperatorsV1alpha1().ClusterServiceVersions(namespace).Get(o.GetName(), metav1.GetOptions{})
+		case *v1.OperatorGroup:
+			fetched, err = client.OperatorsV1().OperatorGroups(namespace).Get(o.GetName(), metav1.GetOptions{})
+		default:
+			require.Failf(t, "couldn't find expected object", "%#v", object)
+		}
+		require.NoError(t, err, "couldn't fetch %s %v", namespace, object)
+		require.True(t, reflect.DeepEqual(object, fetched), diff.ObjectDiff(object, fetched))
+		// require.EqualValues(t, object, fetched, "%s in %s not equal", object.GetObjectKind().GroupVersionKind().String(), namespace)
```

Comment:
Extra commented line here. Also, is there a reason we're directly using reflect instead of `EqualValues`? Diff output is better?

Reply:
I thought so, I feel like it’s a little bit better